lib/mongodb.js

```diff
@@ -191,8 +196,38 @@ MongoDB.prototype.fromDatabase = function(model, data) {
         // Convert the Binary into String
         data[p] = data[p].toString();
       }
+    } else if (data[p] && prop && prop.type && prop.type.name === 'GeoPoint' &&
+      this.settings.enableGeoIndexing === true) {
+      data[p] = {
+        lat: data[p].coordinates[1],
+        lng: data[p].coordinates[0],
+      };
+    }
+  }
+
+  return data;
+};
+
+/*!
+ * Convert JSON to database-appropriate format
+ *
+ * @param {String} model The model name
+ * @param {Object} data The JSON data to convert
+ */
+MongoDB.prototype.toDatabase = function(model, data) {
+  var props = this._models[model].properties;
+
+  for (var p in props) {
+    var prop = props[p];
+    if (data[p] && prop && prop.type && prop.type.name === 'GeoPoint' &&
+      this.settings.enableGeoIndexing === true) {
```

Comment:
Ditto.

In fact, it will be better to extract this long condition into a private method:

```js
    if (this._shouldConvertToCoordinates(data[p], prop)) {
      data[p] = // ...
    }
```

And the implementation:

```js
MongoDB.prototype._shouldConvertToCoordinates = function(data, prop) {
  return data[p] && prop && prop.type && prop.type.name === 'GeoPoint' &&
    this.settings.enableGeoIndexing === true;
};
```